RadixWallet/Clients/DeviceFactorSourceClient/DeviceFactorSourceClient+Live.swift

```diff
@@ -71,30 +71,16 @@ extension DeviceFactorSourceClient: DependencyKey {
 			)
 		}
 
-		let mnemonicMissingEntities: @Sendable () async throws -> (accounts: [AccountAddress], personas: [IdentityAddress]) = {
+		let problematicEntities: @Sendable () async throws -> (mnemonicMissing: ProblematicAddresses, unrecoverable: ProblematicAddresses) = {
 			let deviceFactorSources = try await factorSourcesClient.getFactorSources(type: DeviceFactorSource.self)
 			let entities = try await deviceFactorSources.asyncMap {
 				try await entitiesControlledByFactorSource($0, nil)
 			}
-			.filter { !$0.isMnemonicPresentInKeychain }
 
-			let accounts = entities.flatMap { $0.accounts + $0.hiddenAccounts }.map(\.address)
-			let personas = entities.flatMap { $0.personas + $0.hiddenPersonas }.map(\.address)
+			let mnemonicMissing = entities.filter { !$0.isMnemonicPresentInKeychain }
+			let unrecoverable = entities.filter { !mnemonicMissing.contains($0) && !$0.isMnemonicMarkedAsBackedUp }
```

Comment:
instead of checking every entity to see if it is not contained in `mnemonicMissing` (which is quadratic since `contains` on an array is linear), I would create the intermediate array `mnemonicNotMissing` by running the filter again without the `!`. And then you'd have:
```
let unrecoverable = mnemonicNotMissing.filter { !$0.isMnemonicMarkedAsBackedUp }
```
Not that the quadratic performance will take a lot of time, but still...